sema4ai/vscode-client/src/common.ts

```diff
@@ -164,6 +167,39 @@ export function refreshFilesExplorer() {
     }
 }
 
+export async function revealInExtension(path: string): Promise<void> {
+    const treeView = treeViewIdToTreeView.get(TREE_VIEW_SEMA4AI_TASK_PACKAGES_TREE);
+    if (!treeView.visible) {
+        return;
+    }
+
+    let dataProvider: RobotsTreeDataProvider = treeViewIdToTreeDataProvider.get(
+        TREE_VIEW_SEMA4AI_TASK_PACKAGES_TREE
+    ) as RobotsTreeDataProvider;
+
+    const nodes = await dataProvider.getChildren();
+
+    async function findNode(nodes: RobotEntry[], path: string): Promise<RobotEntry | undefined> {
+        for (const node of nodes) {
+            if (node.robot.directory === path) {
```

Comment:
I think you should also check that the `RobotEntry.type` matches the `RobotEntryType` you expect, just to be sure and maybe `path` should be renamed to `directoryPath`... 

Alternatively you could make the `path` a function that can be called back and then the caller could provide the check function as needed.

-- I'll leave it up to you to choose the approach you prefer ;) 

Also, sometimes the path is not the same on windows, so, `node.robot.directory === path` can fail even if it's the same path, for instance if the listing was done with `C:\temp\bar` but the internal reference is `c:\temp\bar`, so, you should normalize the paths to match on Windows or use a utility function which checks that the paths are the same... there's probably one such utility function available for NodeJS, but you'll need to search to know if that's indeed the case (as this is more costly you should do that only if the `RobotEntryType` matches).

Reply:
Good points, thanks for the feedback. 
I updated the code to use `path.normalize()` to handle OS-specific path differences, and based on my research it's a lightweight operation.